sklearn/datasets/_samples_generator.py

```diff
@@ -942,19 +939,20 @@ def make_blobs(
         for i in range(n_samples % n_centers):
             n_samples_per_center[i] += 1
 
-    for i, (n, std) in enumerate(zip(n_samples_per_center, cluster_std)):
-        X.append(generator.normal(loc=centers[i], scale=std, size=(n, n_features)))
-        y += [i] * n
+    cum_sum_n_samples = np.cumsum(n_samples_per_center)
+    X = np.empty(shape=(sum(n_samples_per_center), n_features), dtype=np.float64)
+    y = np.empty(shape=(sum(n_samples_per_center),), dtype=np.int64)
```

Comment:
the dtype of y currently depends on the system (32 or 64 bits). Let's keep it that way to not break backward compat.
```suggestion
    y = np.empty(shape=(sum(n_samples_per_center),), dtype=int)
```

Reply:
you pushed it before I finished editing it :)

I'm not sure ``int`` will work. Let's see if CI is green. If not, ``np.int_`` will be the right dtype.